src/ResourceManagement/HDInsight/HDInsight/Customizations/ClusterOperations.Create.Customizations.cs

```diff
@@ -570,12 +570,29 @@ private static IEnumerable<Role> GetRoleCollection(ClusterCreateParameters clust
 
             roles.Add(zookeepernode);
 
+            if (clusterCreateParameters.ClusterType.Equals("RServer", StringComparison.OrdinalIgnoreCase))
+            {
+                var EdgeNodeSize = GetEdgeNodeSize(clusterCreateParameters);
+                var EdgeNode = new Role
+                {
+                    Name = "edgenode",
```

Comment:
nit: Why not use ClusterNodeType.EdgeNode.ToString() (or if case matters use .ToLower())
This way strings are not hard-coded and casing can be managed easily.